src/ucs/sys/ucs_netlink.h

```diff
@@ -0,0 +1,54 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2024. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef UCS_NETLINK_H
+#define UCS_NETLINK_H
+
+#include <ucs/type/status.h>
+
+#include <stddef.h>
+#include <linux/netlink.h>
+
+
+#define ucs_netlink_foreach(elem, msg, len) \
+    for (elem = (struct nlmsghdr *)msg; \
+         NLMSG_OK(elem, len) && (elem->nlmsg_type != NLMSG_DONE) && \
+         (elem->nlmsg_type != NLMSG_ERROR); \
+         elem = NLMSG_NEXT(elem, len))
+
+#define ucs_netlink_handle_parse_error(nlh, action) \
+    do { \
+        if (nlh->nlmsg_type == NLMSG_ERROR) { \
+            ucs_diag("failed to parse netlink message header (%d)", \
+                    ((struct nlmsgerr*)NLMSG_DATA(nlh))->error); \
+            action; \
+        } \
+    } while (0)
```

Comment:
maybe check errors inside `ucs_netlink_send_cmd`?

Reply:
@shasson5  Here I check errors for each entry in the routing table so it has to be called from uct_iface.c

Reply:
right but we can add another "pre" iteration over the array, inside netlink infrastructure (because we only have to deal with MSG_ERROR header and not with rtmsg header).